server/diagram-api/src/main/java/io/typefox/sprotty/api/IDiagramServer.java

```diff
@@ -85,4 +91,38 @@
 		IDiagramServer getDiagramServer(String clientId);
 	}
 
+	/**
+	 * A view on the current state of the diagram.
+	 * 
+	 * @author koehnlein
+	 */
+	public interface IDiagramState {
+		/**
+		 * The options received from the client with the last {@link RequestModelAction}. These options
+		 * can be used to control diagram creation. If no such action has been received yet, or the action did
+		 * not contain any options, an empty map is returned.
+		 */
+		Map<String, String> getOptions();
+		
+		/**
+		 * @return the identifier of the client attached to this server.
+		 */
+		String getClientId();
+		
+		/**
+		 * @return the current model
+		 */
+		SModelRoot getCurrentModel();
```

Comment:
Do we need these three getters (options, clientId, model) both in IDiagramServer and in IDiagramState?